tests/components/mqtt/test_init.py

```diff
@@ -776,42 +793,40 @@ async def test_setup_with_tls_config_of_v1_under_python36_only_uses_v1(hass, moc
     assert mock_mqtt.mock_calls[0][2]["tls_version"] == ssl.PROTOCOL_TLSv1
 
 
-async def test_birth_message(hass):
-    """Test sending birth message."""
-    mqtt_client = await async_mock_mqtt_client(
-        hass,
+@pytest.mark.parametrize(
+    "mqtt_config",
+    [
         {
             mqtt.CONF_BROKER: "mock-broker",
             mqtt.CONF_BIRTH_MESSAGE: {
                 mqtt.ATTR_TOPIC: "birth",
                 mqtt.ATTR_PAYLOAD: "birth",
             },
-        },
-    )
+        }
+    ],
+)
+async def test_birth_message(hass, mqtt_client_mock, mqtt_mock):
+    """Test sending birth message."""
     calls = []
-    mqtt_client.publish.side_effect = lambda *args: calls.append(args)
-    hass.data["mqtt"]._mqtt_on_connect(None, None, 0, 0)
+    mqtt_client_mock.publish.side_effect = lambda *args: calls.append(args)
+    mqtt_mock._mqtt_on_connect(None, None, 0, 0)
     await hass.async_block_till_done()
     assert calls[-1] == ("birth", "birth", 0, False)
 
 
-async def test_mqtt_subscribes_topics_on_connect(hass):
+async def test_mqtt_subscribes_topics_on_connect(hass, mqtt_client_mock, mqtt_mock):
     """Test subscription to topic on connect."""
-    mqtt_client = await async_mock_mqtt_client(hass)
-
-    hass.data["mqtt"].subscriptions = [
-        mqtt.Subscription("topic/test", None),
-        mqtt.Subscription("home/sensor", None, 2),
-        mqtt.Subscription("still/pending", None),
-        mqtt.Subscription("still/pending", None, 1),
-    ]
+    await mqtt.async_subscribe(hass, "topic/test", None)
+    await mqtt.async_subscribe(hass, "home/sensor", None, 2)
+    await mqtt.async_subscribe(hass, "still/pending", None)
+    await mqtt.async_subscribe(hass, "still/pending", None, 1)
 
     hass.add_job = MagicMock()
-    hass.data["mqtt"]._mqtt_on_connect(None, None, 0, 0)
+    mqtt_mock._mqtt_on_connect(None, None, 0, 0)
 
     await hass.async_block_till_done()
 
-    assert mqtt_client.disconnect.call_count == 0
+    assert mqtt_client_mock.disconnect.call_count == 0
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5737505)